tests/python/unittest/test_engine.py

```diff
@@ -30,6 +33,44 @@ def test_bulk():
             x += 1
     assert (x.asnumpy() == 104).all()
 
+@unittest.skip("OMP platform dependent")
+def test_engine_openmp_after_fork():
+    """
+    Test that the number of max threads in the child is 1. After forking we should not use a bigger
+    OMP thread pool.
+
+    With GOMP the child always has the same number when calling omp_get_max_threads, with LLVM OMP
+    the child respects the number of max threads set in the parent.
+    """
+    with EnvManager('OMP_NUM_THREADS', '42'):
+        r, w = os.pipe()
+        pid = os.fork()
+        if pid:
+            os.close(r)
+            wfd = os.fdopen(w, 'w')
+            wfd.write('a')
+            omp_max_threads = mx.base._LIB.omp_get_max_threads()
+            print("Parent omp max threads: {}".format(omp_max_threads))
+            try:
+                wfd.close()
+            except:
+                pass
+            try:
+                (cpid, status) = os.waitpid(pid, 0)
+                assert cpid == pid
+                exit_status = status >> 8
+                assert exit_status == 0
+            except:
+                pass
+        else:
+            os.close(w)
+            rfd = os.fdopen(r, 'r')
+            rfd.read(1)
+            omp_max_threads = mx.base._LIB.omp_get_max_threads()
```

Comment:
for our case we need to test GetRecommendedThreadCount. in the parent it should be same as env variable, in child it should be 1. CPP test would be easier to write.

Reply:
C++ test is platform dependent though. There's no fork in windows. That's why I didn't bother with C++.  Do you see it as necessary for approval?

Reply:
C++ test is platform dependent though. There's no fork in windows. That's why I didn't bother with C++. Do you see it as necessary for approval?

Reply:
windows doesnt support pthread_atfork so it should be fine. I think some kind of test is necessary.

Reply:
Added cpp test.